packages/compiler/src/core/helpers/type-name-utils.ts

```diff
@@ -177,12 +177,27 @@ function getInterfaceName(iface: Interface, options: TypeNameOptions | undefined
     interfaceName += `<${iface.templateMapper.args
       .map((x) => getTypeName(x, options))
       .join(", ")}>`;
+  } else if (iface.node.templateParameters.length > 0) {
+    // template
+    const params = iface.node.templateParameters.map((t) => getIdentifierName(t.id.sv, options));
+    interfaceName += `<${params.join(", ")}>`;
   }
   return `${getNamespacePrefix(iface.namespace, options)}${interfaceName}`;
 }
 
 function getOperationName(op: Operation, options: TypeNameOptions | undefined) {
-  return `${getNamespacePrefix(op.namespace, options)}${getIdentifierName(op.name, options)}`;
+  let opName = getIdentifierName(op.name, options);
+  if (isTemplateInstance(op)) {
+    opName += `<${op.templateMapper.args.map((x) => getTypeName(x, options)).join(", ")}>`;
```

Comment:
@timotheeguerin , I follow the same pattern as model here. But I can't figure out when the isTemplateInstance(op) would be true. could you help to describe what's template instance and what's the difference between it and template? thx